ocs_ci/ocs/ocs_upgrade.py

```diff
@@ -340,8 +340,10 @@ def load_version_config_file(self, upgrade_version):
             and config.ENV_DATA["deployment_type"] == "managed"
         )
         use_upstream_mg_image = managed_ibmcloud_platform and not upgrade_in_same_source
-        if (live_deployment and upgrade_in_same_source) or (
-            managed_ibmcloud_platform and not use_upstream_mg_image
+        if (
+            (live_deployment and upgrade_in_same_source)
+            or (managed_ibmcloud_platform and not use_upstream_mg_image)
+            or config["ENV_DATA"].get("platform") == constants.ROSA_HCP_PLATFORM
```

Comment:
@DanielOsypenko hi,

This has introduced an regression in upgrade:

or config["ENV_DATA"].get("platform") == constants.ROSA_HCP_PLATFORM

should be I guess:
```
or config.ENV_DATA.get("platform") == constants.ROSA_HCP_PLATFORM
```
We see error like:
```
2024-09-17 15:11:11          if (
2024-09-17 15:11:11              (live_deployment and upgrade_in_same_source)
2024-09-17 15:11:11              or (managed_ibmcloud_platform and not use_upstream_mg_image)
2024-09-17 15:11:11  >           or config["ENV_DATA"].get("platform") == constants.ROSA_HCP_PLATFORM
2024-09-17 15:11:11          ):
2024-09-17 15:11:11  E       TypeError: 'MultiClusterConfig' object is not subscriptable
2024-09-17 15:11:11  
2024-09-17 15:11:11  ocs_ci/ocs/ocs_upgrade.py:346: TypeError
```